public/app/features/correlations/types.ts

```diff
@@ -30,7 +30,7 @@ type CorrelationConfigType = 'query';
 
 export interface CorrelationConfig {
   field: string;
-  target: object;
+  target: DataLink;
```

Comment:
This isn't a data link. The type is basically "whatever would go into a query editor" which is anything extended off of `DataQuery`, so we need this to be generic.